spanner/google/cloud/spanner_v1/database.py

```diff
@@ -211,8 +211,8 @@ def create(self):
             if exc_to_code(exc.cause) == StatusCode.ALREADY_EXISTS:
                 raise Conflict(self.name)
             elif exc_to_code(exc.cause) == StatusCode.NOT_FOUND:
-                raise NotFound('Instance not found: {name}'.format(
-                    name=self._instance.name,
+                raise NotFound('Database not found: {name}'.format(
```

Comment:
@chemelnucfin 

1. "Instance not found" is the correct error message here (the database is what is being created, so of course it wouldn't be found, it's the parent which might be "not found") You could add more detail like, "Instance {} not found when trying to create database {}".
1. Any reason not to use `exc.cause.details()`?

Reply:
**UPDATE**: I see the example in #4071.

The **issue** is that the **second** DDL statement refers to `yTable` rather than `MyTable` (this seems intentional)?

Thus, returning `exc.cause.details()` seems to be the only right answer here.

In fact `google.api_core.exceptions.from_grpc_error` is made for exactly this case.

Reply:
Everywhere else in the code uses exc_to_code(exc.cause).  I think I will leave it as is.

Reply:
@chemelnucfin That's because this code pre-dates the `api_core` package. Please change it.

Reply:
Can I do it in a separate PR?  There's a lot of other code.

Reply:
The whole point of this PR is to make the exception correct. The correct way to do it is to use the facility provided by `api_core`